neo4j/hierarchy_test.go

```diff
@@ -521,3 +525,165 @@ func TestStore_RemoveRemainMarker_NeoErrExec(t *testing.T) {
 		})
 	})
 }
+
+func TestStore_HierarchyExists(t *testing.T) {
+
+	node := boltstructures.Node{
+		Properties: map[string]interface{}{},
+	}
+	mockNeoResponse := &mapper.Result{
+		Data:  []interface{}{node},
+		Meta:  nil,
+		Index: 0,
+	}
+
+	Convey("Given a bolt connection that returns a node", t, func() {
+		neoDriverMock := &internal.Neo4jDriverMock{
+			ReadWithParamsFunc: func(query string, params map[string]interface{}, mapp mapper.ResultMapper, single bool) error {
+				mapp(mockNeoResponse)
+				return nil
+			},
+		}
+
+		db := &Neo4j{neoDriverMock, 5, 30}
+
+		Convey("When HierarchyExists is called", func() {
+			hierarchyExists, err := db.HierarchyExists(context.Background(), instanceID, dimensionName)
+
+			Convey("Then db.Exec should be called once for the expected query", func() {
+				expectedQuery := fmt.Sprintf(query.HierarchyExists, instanceID, dimensionName)
+				So(len(neoDriverMock.ReadWithParamsCalls()), ShouldEqual, 1)
+				So(neoDriverMock.ReadWithParamsCalls()[0].Query, ShouldEqual, expectedQuery)
+			})
+
+			Convey("Then the return value should be true", func() {
+				So(hierarchyExists, ShouldBeTrue)
+			})
+
+			Convey("Then the returned error should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+		})
+	})
+
+	Convey("Given a bolt connection that returns multiple nodes", t, func() {
+		neoDriverMock := &internal.Neo4jDriverMock{
+			ReadWithParamsFunc: func(query string, params map[string]interface{}, mapp mapper.ResultMapper, single bool) error {
+				mapp(mockNeoResponse)
+				mapp(mockNeoResponse)
+				mapp(mockNeoResponse)
+				return nil
+			},
+		}
+
+		db := &Neo4j{neoDriverMock, 5, 30}
+
+		Convey("When HierarchyExists is called", func() {
+			hierarchyExists, err := db.HierarchyExists(context.Background(), instanceID, dimensionName)
+
+			Convey("Then db.Exec should be called once for the expected query", func() {
+				expectedQuery := fmt.Sprintf(query.HierarchyExists, instanceID, dimensionName)
+				So(len(neoDriverMock.ReadWithParamsCalls()), ShouldEqual, 1)
+				So(neoDriverMock.ReadWithParamsCalls()[0].Query, ShouldEqual, expectedQuery)
+			})
+
+			Convey("Then the return value should be false", func() {
```

Comment:
If multiple nodes are found then the hierarchy does exist, just more than once right?

Reply:
Explicit is good, happy for this not to be changed.